<launch>

    <arg name="width" default="1280"/>
    <arg name="height" default="720"/>
    <arg name="image_format" default="1"/>
    <arg name="enable_ldc_node" default="1"/>
    <arg name="lut_file_path" default="$(find mono_capture)/config/C920_HD_LUT.bin" />
    <arg name="dl_model_path" default="/opt/model_zoo/ONR-OD-8020-ssd-lite-mobv2-mmdet-coco-512x512"/>
    <!-- <arg name="dl_model_path" default="/opt/model_zoo/TFL-OD-2010-ssd-mobV2-coco-mlperf-300x300"/> -->
    <arg name="exportPerfStats" default="0"/>
    <arg name="detVizThreshold" default="0.5"/>
    <!-- video device ID used in /dev/videoX -->
    <arg name="video_id" default="2"/>
    <!-- camera name-->
    <arg name="camera_name" default="camera"/>
    <arg name="framerate" default="30"/>

    <!-- ti_vision_cnn -->
    <node pkg = "ti_vision_cnn" type = "vision_cnn" name = "vision_cnn" output = "screen" args = "" required = "true">
        <rosparam file="$(find ti_vision_cnn)/config/params.yaml" subst_value="true" />
        <param name = "width" value = "$(arg width)"/>
        <param name = "height" value = "$(arg height)"/>
        <param name = "image_format" value = "$(arg image_format)"/>
        <param name = "enable_ldc_node" value= "$(arg enable_ldc_node)"/>
        <param name = "lut_file_path" value = "$(arg lut_file_path)"/>
        <param name = "dl_model_path" value = "$(arg dl_model_path)"/>
        <param name = "input_topic_name" value = "camera/image_raw"/>
        <param name = "rectified_image_topic" value = "camera/image_rect_nv12"/>
        <param name = "rectified_image_frame_id" value = "right_frame"/>
        <param name = "vision_cnn_tensor_topic" value = "vision_cnn/tensor"/>
        <param name = "exportPerfStats" value = "$(arg exportPerfStats)"/>
        <param name = "detVizThreshold" value="$(arg detVizThreshold)"/>
    </node>

    <!-- gscam -->
    <include file="$(find gscam)/launch/v4l_mjpg.launch">
        <arg name="device" value="/dev/video$(arg video_id)" />
        <arg name="node_name" value="gscam$(arg video_id)" />
        <arg name="camera_name" value="$(arg camera_name)" />
        <arg name="framerate" value="$(arg framerate)" />
    </include>
</launch>
